src/transformer/descriptor/descriptor.ts

```diff
@@ -70,8 +70,9 @@ export function GetDescriptor(node: ts.Node, scope: Scope): ts.Expression {
             return GetMethodDeclarationDescriptor(node as ts.MethodDeclaration, scope);
         case ts.SyntaxKind.FunctionType:
             return GetFunctionTypeDescriptor(node as ts.FunctionTypeNode, scope);
+        case ts.SyntaxKind.ConstructSignature:
         case ts.SyntaxKind.CallSignature:
-            return GetFunctionTypeDescriptor(node as ts.CallSignatureDeclaration, scope);
+            return GetFunctionTypeDescriptor(node as ts.CallSignatureDeclaration | ts.ConstructSignatureDeclaration, scope);
```

Comment:
I think you can remove the OR

Reply:
Ah sorry. I didn't see that you may still want to keep the switch case separated. Once you've done that you can remove the OR

Reply:
I put the OR there to make sure the typings are correct... if for some reason we change `GetFunctionTypeDescriptor` to only accept `CallSignatureDeclaration` it would give error... if I remove one of the two then there is no check anymore.

Do you want me to split the 2 cases of the switch?

Reply:
I split the 2 cases